### DISCUSSION TITLE
Editor Gizmo for AABB bounding boxes

### DISCUSSION THREAD

**Discussion by donn-xx** [godotengine/godot-proposals#7839] [Editor]
There are a few issues working in the 3d editor (and I guess the 2d is similar.)

1. You can't see the _size_ of a node other than the scale. (I mean width, height, depth in some unit).
2. You can't see the actual AABB drawn as a box gizmo.
3. It is, in general, hard for me (at least) to reason about geometry and math without being able to see where the AABB is on the screen.

If an AABB gizmo could be created it would solve both of those problems.

Proposed use:
1. You can hide/show the gizmo.
2. When it shows, it also reports the dimensions. In the inspector, or some other place.
4. When you manipulate it, then the Custom AABB is altered.

I think this means including an actual AABB for nodes in general, not only VisualInstances, but even so it would be great!

Thoughts?

**Comment by Calinou** [comment]
> You can't see the size of a node other than the scale. (I mean width, height, depth in some unit).

We could display sizes next to the AABB lines (also in the 2D editor). I had a proof of concept for this in the 2D editor: https://github.com/godotengine/godot-proposals/issues/5288, https://github.com/Calinou/godot/tree/2d-editor-display-selection-size

In 3D, it would look something like this (screenshots from Source 2 Hammer):

![image](https://user-images.githubusercontent.com/180032/268426013-877ca286-6ca7-4cba-b40e-0d4d17805d51.png) | ![image](https://user-images.githubusercontent.com/180032/268426056-77369859-48fe-4355-8f45-be28bb9ac455.png)
-|-

The labels move depending on which parts of the AABB are visible.

> You can't see the actual AABB drawn as a box gizmo.

The orange box outlines you see when selecting a node *is* the default AABB, but it doesn't increase in size when Extra Cull Margin is set above 0 or when a custom AABB is used.

**Reply by donn-xx** [reply]
That looks great. Even a little blender-style 'n-key' panel showing the relevant numbers : postion, center, end, size vector - in local and global coords would be fine. (Probably better since it's hard to read text when the viewport is full of stuff.)

> the orange box
I am slapping my head for not explaining better. I have been working with multimeshes a lot and individual instances within it. I would really like to see those AABBs too. The entire multimesh gets the box, but I'd love some way to see the bboxes for the transform (on an index) within.
